Babylon/commands/abba/run.py

```diff
@@ -0,0 +1,194 @@
+import json
+import pandas as pd
+import plotly.offline as pyo
+import plotly.graph_objs as go
+import plotly.express as px
+from logging import getLogger
+from typing import Any
+from click import argument, command, File
+from Babylon.utils.environment import Environment
+from Babylon.utils.decorators import retrieve_state, injectcontext
+from Babylon.utils.decorators import output_to_file
+from Babylon.utils.credentials import pass_azure_token
+from Babylon.commands.api.scenarios.service.api import ScenarioService
+from Babylon.commands.api.scenarioruns.service.api import ScenarioRunService
+from Babylon.utils.response import CommandResponse
+
+logger = getLogger("Babylon")
+
+env = Environment()
+
+
+def dataframe_to_dict(df: pd.DataFrame, input_types: dict) -> list:
+    """
+    Convert a pandas DataFrame to a list of dictionaries.
+
+    Args:
+      df (pandas.DataFrame): The DataFrame to convert.
+      input_types (dict): A dictionary mapping parameter names to their types.
+
+    Returns:
+      list: A list of dictionaries to be used with the cosmotech-api
+    """
+    result = []
+    for line in df.itertuples():
+        d = dict()
+        d['organizationId'] = line.organizationId
+        d['workspaceId'] = line.workspaceId
+        d['id'] = line.id
+        d['name'] = line.name
+        d['description'] = line.description
+        d['runTemplateId'] = line.runTemplateId
+        if line.scenarioId:
+            d['scenarioId'] = line.scenarioId
+        if line.scenariorunId:
+            d['scenariorunId'] = line.scenariorunId
+        d['parameterValues'] = []
+        for parameter_value in input_types:
+            d['parameterValues'].append({
+                'parameterId': parameter_value,
+                'value': getattr(line, parameter_value),
+                'varType': input_types[parameter_value]
+            })
+        result.append(d)
+    return result
+
+
+@command()
+@injectcontext()
+@pass_azure_token("csm_api")
+@output_to_file
+@argument("input", type=File('r'))
+@argument("var_types", type=File('r'))
+@retrieve_state
+def run(state: Any, azure_token: str, input: str, var_types: str) -> CommandResponse:
+    """Run a series of simulations
+
+    Args:
+      input (str): Table with details of the simulations
+      var_types (str): A table declaring the variable types of columns in input
+    """
+    rows = []  # an array of dicts one for each input row
+    df = pd.read_csv(input, sep='\t')
+    input_types = dict()
+    input_types = {k: v for k, v in (line.rstrip().split("\t") for line in var_types)}
```

Comment:
The same, the declaration on line 73 seems redundant (the same on lines 119-120)